app/controllers/users/sessions_controller.rb

```diff
@@ -218,5 +219,12 @@ def request_id_if_valid
 
       request_id if LETTERS_AND_DASHES.match?(request_id)
     end
+
+    def override_csp_for_google_analytics
+      policy = current_content_security_policy
+      policy.script_src(*policy.script_src, 'dap.digitalgov.gov', 'www.google-analytics.com')
+      policy.connect_src(*policy.style_src, 'www.google-analytics.com')
+      request.content_security_policy = policy
+    end
```

Comment:
what if we exposed this as a `helper` method on `ApplicationController` and then called it from inside the [`dap_analytics.html.erb` template](https://github.com/18F/identity-idp/blob/main/app/views/shared/_dap_analytics.html.erb
) (which is guarded in the base layout)?

https://github.com/18F/identity-idp/blob/ecc6d330eea56465e10b6feabc756d8e727e9c09/app/views/layouts/base.html.erb#L98

That way in case there are other logged out pages outside of this controller, we could still allow?

Reply:
Currently the only `page_with_trust` is the home page, so I'm leaning towards not allowing outside of that.  I don't love the lack of context or hints when it comes to the CSP and controllers since it makes it pretty difficult to track, but I'm not sure I want to go down the road of changing the CSP in a template either.

Reply:
makes sense!